ast/src/parsed/mod.rs

```diff
@@ -86,6 +86,19 @@ pub enum Expression<T, Ref = ShiftedPolynomialReference<T>> {
 }
 
 impl<T, Ref> Expression<T, Ref> {
+    pub fn new_mul(left: Self, right: Self) -> Self {
+        Self::new_binary(left, BinaryOperator::Mul, right)
+    }
+    pub fn new_add(left: Self, right: Self) -> Self {
+        Self::new_binary(left, BinaryOperator::Add, right)
+    }
+    pub fn new_sub(left: Self, right: Self) -> Self {
+        Self::new_binary(left, BinaryOperator::Sub, right)
+    }
```

Comment:
Oh, what I meant to suggest is to implement `std::ops::Add`, etc. It basically has the same interface but I think it would make the code that uses it much more readable!